src/functions.jl

```diff
@@ -303,10 +303,35 @@ end
     return nothing
 end
 
+@inline function one!(c::Taylor1{TaylorN{T}}, k::Int) where {T<:NumberNotSeries}
+    zero!(c, k)
+    if k == 0
+        @inbounds c[0][0][1] = one(constant_term(c[0][0][1]))
+    end
+    return nothing
+end
+
+@inline function identity!(c::HomogeneousPolynomial{T}, a::HomogeneousPolynomial{T}, k::Int) where {T<:NumberNotSeries}
+    @inbounds c[k] = identity(a[k])
+    return nothing
+end
+
 for T in (:Taylor1, :TaylorN)
     @eval begin
-        @inline function identity!(c::$T{T}, a::$T{T}, k::Int) where {T<:Number}
-            @inbounds c[k] = identity(a[k])
+        @inline function identity!(c::$T{T}, a::$T{T}, k::Int) where {T<:NumberNotSeries}
+            if $T == Taylor1
+                @inbounds c[k] = identity(a[k])
+            else
+                @inbounds for l in eachindex(c[k])
+                    identity!(c[k], a[k], l)
+                end
+            end
```

Comment:
This `if` is within a `for` loop, which I guess is bad for performance. I suggest to have it the other way around, though I admit that it implies duplicating the `for` loops in both `if` branches...

Reply:
Thanks for spotting this, I'm now changing the occurrences of this pattern, although this case in particular has the `for` within the `else` so this one is actually correct, right?

Reply:
Ups... sorry